viewer/packages/pointclouds/wasm/src/linalg.rs

```diff
@@ -0,0 +1,78 @@
+use nalgebra_glm::{max2, min2, vec3, vec4, vec4_to_vec3, DMat4, DVec3, DVec4};
+
+#[derive(Clone, Copy, Debug)]
+pub struct BoundingBox {
+    pub min: DVec3,
+    pub max: DVec3,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Vec3WithIndex {
+    pub vec: DVec3,
+    pub index: usize,
+}
+
+impl BoundingBox {
+    pub fn overlaps(&self, other: &BoundingBox) -> bool {
```

Comment:
Think this is possible.
```suggestion
    pub fn overlaps(&self, other: &Self) -> bool {
```